oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/ElasticConnection.java

```diff
@@ -135,8 +162,8 @@ public boolean isAvailable() {
 
     @Override
     public synchronized void close() throws IOException {
-        if (client != null) {
-            client.close();
+        if (hlClient != null) {
```

Comment:
It is slightly weird to check hlClient != null but then use esClient... I think this would be easier to read:

    if (esClient != null) {
        esClient._transport().close();
    }